templates/cookie-notice.php

```diff
@@ -1,58 +1,100 @@
-<div class="options">
-
-<div class="form-item form-checkbox">
-<input id="cookie-accept" type="checkbox" value="1" checked disabled />
-<label for="cookie-accept">Nur Butterkekse
-<br><small>Erforderliche Funktionen</small>
-</label>
-</div>
-<div class="form-item form-checkbox">
-<input id="cookie-accept" type="checkbox" value="1" />
-<label for="cookie-accept">Schokokekse
-<br><small>Statistiken</small>
-</label>
-</div>
-<div class="form-item form-checkbox">
-<input id="cookie-accept" type="checkbox" value="1" />
-<label for="cookie-accept">Space-Cookies
-<br><small>Personalisierung</small>
-</label>
-</div>
-
-</div>
-
-<div class="form-actions">
-<button class="button btn">Meine Auswahl bestätigen</button>
-<button class="button button--primary btn btn--primary">Gönn ich mir alle</button>
-</div>
-
+    <?php if ($options): ?>
+      <div class="options">
+        <?php foreach ($options as $option): ?>
+          <div class="form-item form-checkbox">
+            <label for="<?= esc_attr(sanitize_title($option['name'])) ?>">
+              <input id="<?= esc_attr(sanitize_title($option['name'])) ?>" name="cookies" type="checkbox" value="<?= esc_attr($option['name']) ?>" <?= create_attributes($option['atts'] ?? []); ?> />
+              <?= esc_html($option['title']) ?>
+              <br><small><?= esc_html($option['desc']) ?></small>
+            </label>
+          </div>
+        <?php endforeach; ?>
+      </div>
+    <?php endif; ?>
+    <div class="form-actions">
+      <button class="button btn" data-js="confirm"><?= __('Meine Auswahl bestätigen', Plugin::L10N) ?></button>
+      <button class="button button--primary btn btn--primary" data-js="confirm-all"><?= __('Gönn ich mir alle', Plugin::L10N) ?></button>
+    </div>
   </div>
-  <!--a class="cookie-notice__close" href="#" id="cookies-accept" aria-label="<?= __('OK', Plugin::L10N) ?>"></a-->
 </div>
 
 <script>
 (function () {
-  if (typeof window.localStorage !== 'undefined') {
-    var cookie_notice = document.getElementById('cookie-notice');
-    if (window.localStorage.getItem('cookies-accepted')) {
-      cookie_notice.setAttribute('hidden', 'true');
-      return false;
-    }
-    document.getElementById('cookies-accept').addEventListener('click', function (e) {
-      e.preventDefault();
-      e.stopPropagation();
-      window.localStorage.setItem('cookies-accepted', 'true');
-      cookie_notice.setAttribute('hidden', 'true');
+  if (typeof window.localStorage === 'undefined') {
+    return;
+  }
+
+  function hideCookieNotice() {
+    const cookie_notice = document.getElementById('cookie-notice');
+    cookie_notice.setAttribute('hidden', 'true');
+    document.body.classList.remove('has-cookie-notice');
+  }
+
+  function DOMContentLoaded() {
+    document.body.classList.add('has-cookie-notice');
+    document.addEventListener('click', (event) => {
```

Comment:
We should be careful with backwards-compatibility with the modal dialog now, because the user wouldn't be able to dismiss it.  But we can probably control that more granularly when compiling the script anyway.